src/main/java/com/michalsydoryk/app/board/Board2D.java

```diff
@@ -0,0 +1,87 @@
+package com.michalsydoryk.app.board;
+
+import com.michalsydoryk.app.coordinates.Coordinates2D;
+import com.michalsydoryk.app.coordinatescomparator.Coordinates2DComparator;
+import com.michalsydoryk.app.sign.Sign;
+
+import java.util.Set;
+import java.util.TreeMap;
+
+public final class Board2D implements Board<Coordinates2D> {
+    final TreeMap<Coordinates2D, Sign> fields;
```

Comment:
..and here we are, implementations are also public. So the "board" package - what is it for?